File: backend/directory_manager.py
import os

class DirectoryManager:
    def __init__(self, base_root="book_data"):
        self.base_root = base_root

    def get_target_path(self, task_type, sub_folder):
        # Creates: book_data/College_Book/Chapter_1
        path = os.path.join(self.base_root, task_type, sub_folder)
        os.makedirs(path, exist_ok=True)
        return path

dm = DirectoryManager()

File: backend/directory_manager.js
import fs from 'fs';
import path from 'path';

export const directoryManager = {
    getTargetPath(task, subfolder) {
        const baseRoot = 'book_data';
        const targetPath = path.join(baseRoot, task, subfolder);
        if (!fs.existsSync(targetPath)) {
            fs.mkdirSync(targetPath, { recursive: true });
        }
        return targetPath;
    }
};